nomad/upload_metrics.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+
+# this script is executed on Nomad clients, after a scenario has been run, to upload metrics with telegraf.
+# It also takes care of setting the RUN_ID if unset from env
+
+set -euo pipefail
+
+function check_envset() {
+  local var_name="$1"
+  # temporarily allow fail to bind
+  # Do not use `[[ -v $var_name ]]` as it is not compatible with MacOS bash (still using 3.2)
+  set +u
+  if [[ -z "${!var_name}" ]]; then
+    set -u
+    echo "Environment variable $var_name is not set or is an empty string." >&2
+    exit 1
+  fi
+  set -u
+}
+
+check_envset "WT_METRICS_DIR"
+check_envset "INFLUX_TOKEN"
+check_envset "TELEGRAF_CONFIG_PATH"
+
+# read RUN_ID if not set in the environment
+if [ -z "$RUN_ID" ]; then
+    # if isset RUN_SUMMARY_PATH
+    summary_path="run_summary.jsonl"
+    if [ ! -z "$RUN_SUMMARY_PATH" ]; then
+        summary_path="$RUN_SUMMARY_PATH"
+    fi
+
+    if [ -f "$summary_path" ]; then
+        # NOTE: nomad workspace doesn't have jq installed, so we can't use it
+        RUN_ID=$(grep -o '"run_id":"[^"]*"' "$summary_path" | sed 's/.*:"\([^"]*\)".*/\1/' | tail -n 1)
```

Comment:
jq indeed doesn't work on nomad. This is the only command combination I could find.